ledger-core/virtual/object/finalizedstate/statefinalizer.go

```diff
@@ -69,6 +69,7 @@ func (sm *SMStateFinalizer) stepSendVStateReport(ctx smachine.ExecutionContext)
 
 	msg := sm.Report
 
+	msg.ProvidedContent.LatestValidatedState = msg.ProvidedContent.LatestDirtyState
```

Comment:
I think that it is a bad place to set ValidatedState.
Maybe better do it in SMObject (see `BuildLatestDirtyState()` and `BuildStateReport()` for reference)?\

I mean that in future LatestValidatedState will differ from LatestDirtyState.

Reply:
Fixed